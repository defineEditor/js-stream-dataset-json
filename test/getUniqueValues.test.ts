import DatasetJson from '../src/index';

test('Get unique values of variables', async () => {
    const filePath = 'test/data/adsl.json';

    const data = new DatasetJson(filePath);
    const values = await data.getUniqueValues({columns: ['USUbjid', 'trtedt', 'STUDYID'], limit: 20});
    expect(values).toMatchSnapshot();
});

test('Get unique values of variables and do not sort the result', async () => {
    const filePath = 'test/data/adsl.json';

    const data = new DatasetJson(filePath);
    const values = await data.getUniqueValues({columns: ['USUbjid', 'trtedt', 'STUDYID'], limit: 20, bufferLength: 10, sort: false});
    expect(values).toMatchSnapshot();
});

test('Get unique values with counts', async () => {
    const filePath = 'test/data/adsl.json';

    const data = new DatasetJson(filePath);
    const values = await data.getUniqueValues({columns: ['USUbjid', 'trtedt', 'STUDYID', 'BMIBL'],
        limit: 10, addCount: true, sort: false});
    expect(values).toMatchSnapshot();
});