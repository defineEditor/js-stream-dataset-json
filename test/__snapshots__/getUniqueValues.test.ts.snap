// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get unique values of variables 1`] = `
{
  "STUDYID": {
    "counts": {},
    "values": [
      "CDISCPILOT01",
    ],
  },
  "TRTEDT": {
    "counts": {},
    "values": [
      "2012-09-01",
      "2012-09-16",
      "2013-01-20",
      "2013-01-23",
      "2013-03-09",
      "2013-03-18",
      "2013-03-24",
      "2013-04-28",
      "2013-06-26",
      "2013-08-03",
      "2013-12-09",
      "2014-01-14",
      "2014-02-20",
      "2014-03-16",
      "2014-03-31",
      "2014-07-02",
      "2014-07-09",
      "2014-08-16",
      "2014-09-09",
      "2014-12-30",
    ],
  },
  "USUBJID": {
    "counts": {},
    "values": [
      "01-701-1015",
      "01-701-1023",
      "01-701-1028",
      "01-701-1033",
      "01-701-1034",
      "01-701-1047",
      "01-701-1097",
      "01-701-1111",
      "01-701-1115",
      "01-701-1118",
      "01-701-1130",
      "01-701-1133",
      "01-701-1146",
      "01-701-1148",
      "01-701-1153",
      "01-701-1180",
      "01-701-1181",
      "01-701-1188",
      "01-701-1192",
      "01-701-1203",
    ],
  },
}
`;

exports[`Get unique values of variables and do not sort the result 1`] = `
{
  "STUDYID": {
    "counts": {},
    "values": [
      "CDISCPILOT01",
    ],
  },
  "TRTEDT": {
    "counts": {},
    "values": [
      "2014-07-02",
      "2012-09-01",
      "2014-01-14",
      "2014-03-31",
      "2014-12-30",
      "2013-03-09",
      "2014-07-09",
      "2012-09-16",
      "2013-01-23",
      "2014-09-09",
      "2014-08-16",
      "2013-04-28",
      "2013-06-26",
      "2014-02-20",
      "2014-03-16",
      "2013-03-18",
      "2013-12-09",
      "2013-03-24",
      "2013-01-20",
      "2013-08-03",
    ],
  },
  "USUBJID": {
    "counts": {},
    "values": [
      "01-701-1015",
      "01-701-1023",
      "01-701-1028",
      "01-701-1033",
      "01-701-1034",
      "01-701-1047",
      "01-701-1097",
      "01-701-1111",
      "01-701-1115",
      "01-701-1118",
      "01-701-1130",
      "01-701-1133",
      "01-701-1146",
      "01-701-1148",
      "01-701-1153",
      "01-701-1180",
      "01-701-1181",
      "01-701-1188",
      "01-701-1192",
      "01-701-1203",
    ],
  },
}
`;

exports[`Get unique values with counts 1`] = `
{
  "BMIBL": {
    "counts": {
      "21.9": 1,
      "23.9": 2,
      "25.1": 1,
      "26.1": 1,
      "27.3": 1,
      "27.6": 1,
      "28.1": 1,
      "28.8": 1,
      "30.4": 2,
      "31.4": 1,
    },
    "values": [
      25.1,
      30.4,
      31.4,
      28.8,
      26.1,
      27.3,
      23.9,
      21.9,
      27.6,
      28.1,
    ],
  },
  "STUDYID": {
    "counts": {
      "CDISCPILOT01": 254,
    },
    "values": [
      "CDISCPILOT01",
    ],
  },
  "TRTEDT": {
    "counts": {
      "2012-09-01": 1,
      "2012-09-16": 1,
      "2013-01-23": 1,
      "2013-03-09": 1,
      "2014-01-14": 1,
      "2014-03-31": 1,
      "2014-07-02": 1,
      "2014-07-09": 1,
      "2014-09-09": 1,
      "2014-12-30": 1,
    },
    "values": [
      "2014-07-02",
      "2012-09-01",
      "2014-01-14",
      "2014-03-31",
      "2014-12-30",
      "2013-03-09",
      "2014-07-09",
      "2012-09-16",
      "2013-01-23",
      "2014-09-09",
    ],
  },
  "USUBJID": {
    "counts": {
      "01-701-1015": 1,
      "01-701-1023": 1,
      "01-701-1028": 1,
      "01-701-1033": 1,
      "01-701-1034": 1,
      "01-701-1047": 1,
      "01-701-1097": 1,
      "01-701-1111": 1,
      "01-701-1115": 1,
      "01-701-1118": 1,
    },
    "values": [
      "01-701-1015",
      "01-701-1023",
      "01-701-1028",
      "01-701-1033",
      "01-701-1034",
      "01-701-1047",
      "01-701-1097",
      "01-701-1111",
      "01-701-1115",
      "01-701-1118",
    ],
  },
}
`;
